OptionsSampleApp/app/src/main/java/com/ooyala/sample/players/AbstractHookActivity.java

```diff
@@ -0,0 +1,134 @@
+package com.ooyala.sample.players;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+import com.ooyala.android.OoyalaNotification;
+import com.ooyala.android.OoyalaPlayer;
+import com.ooyala.android.ui.OoyalaPlayerLayoutController;
+import com.ooyala.android.ui.OptimizedOoyalaPlayerLayoutController;
+import com.ooyala.android.util.SDCardLogcatOoyalaEventsLogger;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+
+abstract class AbstractHookActivity extends Activity implements Observer {
+
+	private static final int PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE = 1;
+	protected String TAG = this.getClass().toString();
+
+	private SDCardLogcatOoyalaEventsLogger log = new SDCardLogcatOoyalaEventsLogger();
+	private String text;
+	protected OptimizedOoyalaPlayerLayoutController optimizedOoyalaPlayerLayoutController;
+	protected OoyalaPlayerLayoutController playerLayoutController;
```

Comment:
What happens if the activity owning the `playerLayoutController` instance calls `onStop` or `onPause`? It doesn't look like the video will be paused and resumed as you do for `optimizedOoyalaPlayerLayoutController`. Test those use cases and change if there's an issue.